### PR TITLE
Extension detection

### PR DESCRIPTION
**Pull request by SamueLagunju**

### CODE REVIEW

File: crypto/__main__.py [hunk 1]
```diff
@@ -61,6 +66,24 @@ def main():
             else:
                 print("Input {0} was not a valid file.".format(file_name))
 
+    # NEW - File encryption based on its extension
+    elif files:
+        for filename in files:
+            crypter = Crypter(filename, SeanStrategy())
+            file_contents = read_file(filename)
+            if any(file in filename for file in decryption_extension):
```
Comment (KaiPrince) [inline crypto/__main__.py:74]:
```suggestion
            if any(ext in filename for ext in decryption_extension):
```

for clarity.

File: crypto/__main__.py [hunk 2]
```diff
@@ -61,6 +66,24 @@ def main():
...
+                print("Decrypting: {0}".format(filename))
+                output_text = crypter.decrypt_txt(file_contents)
+            if any(file in filename for file in encryption_extension):
```
Comment (KaiPrince) [inline crypto/__main__.py:77]:
```suggestion
            elif any(file in filename for file in encryption_extension):
```
these two cases should be mutually exclusive.

Comment (KaiPrince) [inline crypto/__main__.py:77]:
```suggestion
            if any(ext in filename for ext in encryption_extension):
```

File: crypto/__main__.py [hunk 3]
```diff
@@ -61,6 +66,24 @@ def main():
...
+                print("Encrypting: {0}".format(filename))
+                output_text = crypter.encrypt_txt(file_contents)
+            write_file(filename, output_text)
+            check_write(output_text, filename)
+            new_file = convert_ext(filename)
```
Comment (KaiPrince) [inline crypto/__main__.py:82]: The output is written to `filename`, but then the a new file name is made by `convert_ext`, which is then never used.
Is `write_file` writing to the wrong file?

**Review by KaiPrince — CHANGES_REQUESTED** [review]

**Review by SamueLagunju — COMMENTED** [review]
This looks good to me.

**Pull request merged**